### DISCUSSION TITLE
How to dynamically change the content of a QTable cell?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#13130] [General - Components / Directives / etc] [answered]
I have a QTable and one of the columns is defined as 

    {name: 'author', field: (row: Case) => row.Author, label: 'Author', align: 'left', sortable: true},

`row.Author` is a string.

I need to `fetch` an API with that value to get enriched information (say, `row.Author` is an id, and by calling the API I can have the full name). This is an asynchronous operation.

**Is there a way to do this with QTable?**

In "typical Vue" I would create a reactive variable that would be updated as part of the `fetch` call but I cannot see how to fit this in a QTable cell.

**Comment by metalsadman** [comment, marked as answer]
the question is when would you fetch the call? thru event? etc...

if it were me, i would add an extra property in the data that will hold the value when the api call resolves, then display it using slot. like so:
```html
<template #body-cell-author="props">
  <q-td :props="props" @click.stop="callSomething(props)">
    <div v-if="props.row.placeHolder">
       {{pros.row.placeHolder}}
    </div>
  </q-td>
</template>
```
```js
...
   async function callSomething(props) {
     props.row.placeHolder = await myApiCall('/someUrl', props.row.Author)
   }
...
```
